src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -330,7 +333,17 @@ public VAppTemplate getImage(final String imageId) {
 
    @Override
    public Iterable<Vm> listNodes() {
-      return Sets.newHashSet();
+      Set<Vm> vms = Sets.newHashSet();
```

Comment:
Personal preference for `newLinkedHashSet` as gives more determinism - same order as they are returned to us by the vcloud api.